src/components/implementation/no_interface/scheddev/sl.c

```diff
@@ -61,21 +83,66 @@ sl_thd_block(thdid_t tid)
 
 	sl_cs_enter();
 	t = sl_thd_curr();
-	if (unlikely(t->state == SL_THD_WOKEN)) {
-		t->state = SL_THD_RUNNABLE;
+	if (sl_thd_block_no_cs(t, 0)) {
```

Comment:
Yes, `block_type` seems to be much better. My main concern is passing `thd_state` as a parameter to this function, it is kinda needing the caller to know the `thd_states`. Instead I chose this.